### PR TITLE
Improve performance of include_fields filter

### PR DESCRIPTION
**Pull request by Elkasitu**
This PR introduces changes to increase the performance of the `include_fields` filter by leveraging `QuerySet.prefetch_related()` and `QuerySet.only()`.

Before this commit, a query such as `/flaws?include_fields=cve_id` with 10 concurrent users would yield about 5.3 requests per second and have an average response time of 2180ms

image

After this commit, the same query with 10 concurrent users yields 10 requests per second and has an average response time of 179ms.

image

In practice this means that for 10 concurrent users, there is no bottleneck for this particular query, on top of that it means that the performance gains are of around an order of magnitude in the best-case scenario.

The max throughput was also tested, and with 100 concurrent users (unrealistic scenario) the average requests per second is about 30 and the average response time 1200ms, which is manageable.

image

I believe that at 100 users, the bottleneck is mostly at the CPU/RAM level and could be further improved by scaling horizontally, meaning that code-wise the bottlenecks are unlikely.

### CODE REVIEW

**Review by osoukup — APPROVED** [review]
Great improvements! Well documented too. LGTM

File: etc/pg/postgresql.conf [hunk 1]
```diff
@@ -61,7 +61,7 @@ listen_addresses = '*'		# what IP address(es) to listen on;
 					# defaults to 'localhost'; use '*' for all
 					# (change requires restart)
 #port = 5432				# (change requires restart)
-max_connections = 100			# (change requires restart)
+max_connections = 200			# (change requires restart)
```
Comment (JimFuller-RedHat) [inline etc/pg/postgresql.conf:64]: Informational - osidb stage/prod are either 3000/5000 max conns ... though it is good to constrain this further for local development

**Review by JimFuller-RedHat — APPROVED** [review]
nice and LGTM

File: osidb/tests/endpoints/flaws/test_flaws.py [hunk 2]
```diff
@@ -778,6 +778,32 @@ def test_list_flaws_include_and_exclude_fields(self, auth_client, test_api_uri):
...
+        assert body["count"] == 0
+
+        affect = AffectFactory()
+        AffectFactory(flaw=affect.flaw)
+
+        response = auth_client().get(f"{test_api_uri}/flaws?include_fields=affects")
+        assert response.status_code == 200
+        body = response.json()
+        assert body["count"] == 1
+        # only one key should be present, affects"
```
Comment (jobselko) [inline osidb/tests/endpoints/flaws/test_flaws.py:803]: Redundant `"`

**Review by jobselko — APPROVED** [review]
LGTM, great job!

File: osidb/filters.py [hunk 3]
```diff
@@ -70,7 +71,94 @@ def __init__(self, data=None, queryset=None, *, request=None, prefix=None):
...
+                continue
+            field_set.add(fname)
+        return prefetch_set, field_set
+
+    def include_fields_filter(self, queryset, name, value):
+        """
+        Optimizes a view's QuerySet based on user input.
+
+        This filter will attempt to optimize a given view's queryset based on an
+        allowlist of fields provided by the user in order to improve performance.
```
[the remark this answers is not in the document]
Comment (Elkasitu) [reply osidb/filters.py:126]: I will but this is essentially knowledge of the django-filters API

**Pull request merged**